src/lib/services/query-service.ts

```diff
@@ -0,0 +1,47 @@
+import type { WorkflowExecution } from '$lib/models/workflow-execution';
+import { requestFromAPI } from '$lib/utilities/request-from-api';
+
+type StackTraceOptions = {
+  workflow: WorkflowExecution;
+  namespace: string;
+};
+
+type QueryPayload = {
+  data: string;
+  metadata: {
+    encoding: string;
+  };
+};
+
+type QueryType = {
+  payloads: QueryPayload[];
+};
+
+type StackTraceExecution = {
+  queryRejected: string | null;
```

Comment:
Do you ever intend for it to be `null` or just `undefined`? If not, you can just say `string` here and keep it optional.